src/rpdk/core/contract/suite/handler_commons.py

```diff
@@ -154,8 +154,23 @@ def test_input_equals_output(resource_client, input_model, output_model):
     pruned_output_model = prune_properties_if_not_exist_in_path(
         output_model.copy(), pruned_input_model, resource_client.read_only_paths
     )
+
     pruned_output_model = prune_properties_if_not_exist_in_path(
         pruned_output_model, pruned_input_model, resource_client.create_only_paths
     )
 
-    assert pruned_input_model == pruned_output_model
+    assertion_error_message = (
+        "All properties specified in the request MUST "
+        "be present in the model returned, and they MUST"
+        " match exactly, with the exception of properties"
+        " defined as writeOnlyProperties in the resource schema"
```

Comment:
think this error message might be misleading because there should be more exceptions:
[readOnlyProperties, writeOnlyProperties, defaults, insertionOrder, uniqueItems, etc.](https://github.com/aws-cloudformation/aws-cloudformation-resource-schema/#resource-semantics)